lte/gateway/python/magma/mobilityd/ip_address_man.py

```diff
@@ -123,49 +124,15 @@ def __init__(self,
 
                 Default: None, no recycling will occur automatically.
         """
-
-        persist_to_redis = config.get('persist_to_redis', True)
-        redis_port = config.get('redis_port', 6379)
-
-        self.allocator_type = allocator_type
-        logging.debug('Persist to Redis: %s', persist_to_redis)
         self._lock = threading.RLock()  # re-entrant locks
 
         self._recycle_timer = None  # reference to recycle timer
         self._recycling_interval_seconds = recycling_interval
 
-        if not persist_to_redis:
-            self._assigned_ip_blocks = set()  # {ip_block}
-            self.sid_ips_map = defaultdict(IPDesc)  # {SID=>IPDesc}
-            self._dhcp_gw_info = UplinkGatewayInfo(defaultdict(str))
-
-        else:
-            if not redis_port:
-                raise ValueError(
-                    'Must specify a redis_port in mobilityd config.')
-            client = get_default_client()
-            self._assigned_ip_blocks = store.AssignedIpBlocksSet(client)
-            self.sid_ips_map = store.IPDescDict(client)
-            self._dhcp_gw_info = UplinkGatewayInfo(store.GatewayInfoMap())
-
-        self.ip_state_map = IpDescriptorMap(persist_to_redis, redis_port)
-        logging.info("Using allocator: %s", self.allocator_type)
-
-        if self.allocator_type == MobilityD.IP_POOL:
-            self._dhcp_gw_info.read_default_gw()
-            self.ip_allocator = IpAllocatorStatic(self._assigned_ip_blocks,
-                                                  self.ip_state_map,
-                                                  self.sid_ips_map)
-        elif self.allocator_type == MobilityD.DHCP:
-            dhcp_store = store.MacToIP()  # mac => DHCP_State
-            iface = config.get('dhcp_iface', 'dhcp0')
-            retry_limit = config.get('retry_limit', 300)
-            self.ip_allocator = IPAllocatorDHCP(self._assigned_ip_blocks,
-                                                self.ip_state_map,
-                                                iface=iface,
-                                                retry_limit=retry_limit,
-                                                dhcp_store=dhcp_store,
-                                                gw_info=self._dhcp_gw_info)
+        self.ip_allocator = ip_allocator_factory.get_allocator()
+        self.sid_ips_map = ip_allocator_factory.get_sid_ips_map()
+        self.ip_state_map = ip_allocator_factory.get_ip_states_map()
+        self._dhcp_gw_info = ip_allocator_factory.get_dhcp_gw_info()
```

Comment:
https://github.com/magma/magma/wiki/Contributing:-General-Code-Conventions#only-write-trivial-logic-in-contructors

You've put the complex initialization logic in a different file but it's still being run in this constructor. You should inject the individual state stores instead of the factory in the constructor of this component. Determining what the stores are backed by (memory maps or Redis) should be left to the main.py for mobilityd